### DISCUSSION TITLE
Creating ephemeral applications

### DISCUSSION THREAD

**Discussion by rdelacruz2** [argoproj/argo-cd#9281] [Q&A]
Hello!

Right now we have plans to start creating what we called "ephemeral environments"...the idea is to provide to the developer a better experience when testing new changes.

Right now, all the applications are living in the same ArgoCD instance, we have another ArgoCD instance that we'll use to deploy this new ephemeral stack.

Our app structure is the following:

#ArgoCD repo
- One definition of one application that will use a helm chart repository
- This helm chart will deploy more applications inside the ArgoCD cluster
- Adding values of the applications and image to use (we are using the commit to identify the image and the changes in the repo)
- Apps of apps arch
- Inside the repository of the app, we have folders per env (with all the variables) using Kustomize structure

A summary:
A mix between helm chart that deploys application that uses kustomize

The goal:

The idea is to have:

application: ephemeral-application-2342dfgba
deploy inside all the dependencies overriding the default values (for example change the URL of the VirtualService to reach the new one without impacting the rest)

Have anyone experienced something like this?

Best regards!

**Comment by jayunit100** [comment]
this is an interesting question, and currently, the only alternative is to

1) Using a CRD with a dynamic depoyment strategy.  I suppose one approach *could* be that you could have an "argo operator" which created (and merged) a GH PR when a CRD was created, and then deleted that PR when a CRD was deleted. 

2) Using `ApplicationSets` for this functionality, as it's probably the closest thing to a "You don't have to build an operator" approach.
